src/main/java/gregtech/api/multiblock/BlockPatternChecker.java

```diff
@@ -0,0 +1,49 @@
+package gregtech.api.multiblock;
+
+import codechicken.lib.vec.Vector3;
+import gregtech.api.metatileentity.multiblock.MultiblockControllerBase;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+
+public class BlockPatternChecker {
+    
+    public static BlockPos getPatternErrorPos(MultiblockControllerBase controllerBase) {
+        BlockPattern structurePattern = controllerBase.getStructurePattern();
+        if(structurePattern != null && structurePattern.checkPatternAt(controllerBase.getWorld(), controllerBase.getPos(), controllerBase.getFrontFacing().getOpposite()) == null) {
+            return new BlockPos(structurePattern.getBlockPos());
+        }
+        return null;
+    }
+
+    public static PatternMatchContext checkPatternAt(MultiblockControllerBase controllerBase) {
+        BlockPattern structurePattern = controllerBase.getStructurePattern();
+        return structurePattern == null ? null : structurePattern.checkPatternAt(controllerBase.getWorld(), controllerBase.getPos(), controllerBase.getFrontFacing().getOpposite());
+    }
+
+    public static EnumFacing getSpin(MultiblockControllerBase controllerBase) {
```

Comment:
Not sure how you are going to override a static method? :-)

Reply:
i was going to write a coremod directly . If you don't want too much abstraction in GTCE.

Reply:
An abstraction that is understandable for future maintainers is better than a coremod hack. :-)

I guess this can just be an extra method in MultiblockControllerBase for people to override.

Reply:
yeah, I'd be happy to do that too.